src/libraries/System.Private.CoreLib/src/System/Reflection/InvokerEmitUtil.cs

```diff
@@ -196,6 +210,14 @@ public static unsafe InvokeFunc_RefArgs CreateInvokeDelegate_RefArgs(MethodBase
             return (InvokeFunc_RefArgs)dm.CreateDelegate(typeof(InvokeFunc_RefArgs), target: null);
         }
 
+        private static void EmitTrueNullableUnbox(ILGenerator il, Type parameterType)
+        {
+            // Unbox the true Nullable<T> created by reflection to a Nullable<T> without using
+            // OpCodes.Unbox since unboxing that is not necessarily a valid CLI operation.
+            il.Emit(OpCodes.Call, Methods.Object_GetRawData());
+            il.Emit(OpCodes.Ldobj, parameterType);
```

Comment:
We do all type safety checks in the non-dynamic managed code.

Can you use this sequence for all pointers and valuetypes? It would skipt the type check for all them, just like we are skipping the type check for nullable here.

Reply:
Hmm I applied those changes to the other args but I think there may be a problem in general when invoke is used to call a method on a `Nullable<T>` -- this is based on changing the other .Unbox used for the `this` pointer to the new helper method where I saw some test failures. I'll update based on what I find.